### ISSUE TITLE
Simplify Shipit Model

### ISSUE THREAD

**Issue by jkurz** [turnerlabs/harbor#34]
We can still create a simplified shipit model in shipit. We could possibly do this without breaking anything as well. 

1. create the new model
2. update apis to translate current objects into new model
3. the new model could store a field, `from: "environment"` which would mean this envVar came from the environment level.
3. create /v2/ path, so we can update the new shipit model directly. 
4. on the fly GETs to /v1/ path could translate the new model back to old so we don't break any clients as we rollout. 
5. then we could work on updating all of the clients
6. monitor logs to see if anyone is hitting /v1/ paths, if not, we could remove logic to translate GETs

Doing it this way, would not break any applications, but yet we would have a much more simplified model.

**Comment by wilsonodk** [comment]
I was also thinking we could start moving all Env Vars to Container Env Var in prep for something like this. We'd first have to make sure everything reads off Container Env Var. But, then we could revamp the clients to just treat all Env Vars as Container Env Vars. For example the Env Vars tab would just show the Env Vars that are set on each Container (we'd of course remove them from the Containers tab).

But, I think we can take a slow, methodical approach to achieve what we wanted with ShipIt v3. It'll just take several iterations.

**Comment by jkurz** [comment]
Agreed 100%. It will take time. I think moving things to the container level in the UI is a great first step. This task is definitely not something that can be done in one PR. Maybe we should create some smaller issues on the shipit-api and harbor-ui repo and link back here to keep track of what we are doing to accomplish this.

or is this something we even need to worry about right now? I mean it does not necessarily affect our customers, and there are many other things we could spend our time on.
